certbot-ci/certbot_integration_tests/assets/bind-config/conf/named.conf

```diff
@@ -0,0 +1,41 @@
+
+options {
+  directory "/var/cache/bind";
+
+  // Running inside Docker. Bind address on Docker host is 127.0.0.1.
+  listen-on { any; };
+  listen-on-v6 { any; };
+
+  // Not allowing recursion to prevent query leaks to the internet.
+  // This should be okay for Pebble because it only makes queries to the authoritative
+  // zone, but perhaps not for Boulder which will do e.g. CAA tree climbing.
+  // In the latter case, we might need to write an RPZ to do some local responses.
```

Comment:
Maybe we could solve this by adding a forward server to a non-existent DNS server. Even better, maybe we could still start a `challtestsrv` instance as configured currently, and make bind9 set a forwarder to it. This way, every request is ensured to always return localhost eventually.

To would certainly need a redefinition of how `challtestsrv` is set, because currently it is an external component for Pebble, and an internal one for Boulder, making it harder to instrumentate. However doing so would simplify a bit the way CA servers are set up (by merging the logic of the DNS mock server), so I do not think that resolving the issue here will significantly increase the complexity.